mlflow/server/handlers.py

```diff
@@ -131,19 +136,35 @@ def _get_sqlalchemy_store(cls, store_uri):
 
 def _get_artifact_repo_mlflow_artifacts():
     """
-    Get an artifact repository specified by `--artifacts-destination` option for `mlflow server`
+    Get an artifact repository specified by ``--artifacts-destination`` option for ``mlflow server``
     command.
     """
     from mlflow.server import ARTIFACTS_DESTINATION_ENV_VAR
 
     global _artifact_repo
-
     if _artifact_repo is None:
         _artifact_repo = get_artifact_repository(os.environ[ARTIFACTS_DESTINATION_ENV_VAR])
-
     return _artifact_repo
 
 
+def _get_serve_artifacts_mode():
+    """
+    Initializes ``_serve_artifacts_mode`` that indicates whether the MLflow server is operating in
+    ``-serve_artifacts`` mode for artifact listing and returns the value.
+    """
+    from mlflow.server import SERVE_ARTIFACTS_ENV_VAR
+
+    global _serve_artifacts_mode
+    if _serve_artifacts_mode is None:
+        environment_flag = os.environ.get(SERVE_ARTIFACTS_ENV_VAR, "false") == "true"
+        artifact_repo = _get_artifact_repo_mlflow_artifacts()
+        _serve_artifacts_mode = environment_flag and isinstance(
+            artifact_repo,
+            (HttpArtifactRepository, MlflowArtifactsRepository, LocalArtifactRepository),
+        )
```

Comment:
Do we need to check this? When handling a ListArtifacts or `get-artifact` request, the MLflow server should only resolve a proxied artifact URI to an underlying location and use that underlying location to serve requests if the following are true:

1. The server is configured to run in proxied mode (i.e. the `SERVE_ARTIFACTS_ENV_VAR` os env var is set to `true`)

2. The specified run has a proxied artifact URI. We don't seem to be checking this here; instead, we're checking the artifact repo type associated with the configured artifact destination.